### PR TITLE
SUMO-170423: Create user data source

### PR DESCRIPTION
**Pull request by troy-q-li**
Created data source for user.

### CODE REVIEW

**Comment by troy-q-li** [comment]
Commit too many times for debugging and testing this time. Not the best practice and will avoid it in the future.

File: sumologic/data_source_sumologic_user.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		},
+	}
+}
+
+func dataSourceSumologicUserRead(d *schema.ResourceData, meta interface{}) error {
+	c := meta.(*Client)
+
+	var user *User
+	var err error
+	if uid, ok := d.GetOk("id"); ok {
```
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:56]: `uid` -> `userId` ?

File: sumologic/data_source_sumologic_user.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		id := uid.(string)
+		user, err = c.GetUser(id)
+		if err != nil {
+			return fmt.Errorf("user with id %v not found: %v", id, err)
+		}
+	} else {
+		if uemail, ok := d.GetOk("email"); ok {
```
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:63]: `uemail` -> `userEmail`?

File: sumologic/data_source_sumologic_user.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	d.Set("is_active", user.IsActive)
+	if err := d.Set("role_ids", user.RoleIds); err != nil {
+		return fmt.Errorf("error setting role ids for datasource %s: %s", d.Id(), err)
+	}
+
+	log.Printf("[DEBUG] data_source_sumologic_user: retrieved %v", user)
+	return nil
+}
+
+func (s *Client) GetUserEmail(email string) (*User, error) {
```
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:90]: We should call this method `GetUserByEmail`.

File: website/docs/d/user.html.markdown [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+layout: "sumologic"
+page_title: "SumoLogic: sumologic_user"
+description: |-
+Provides a way to retrieve Sumo Logic user details (id, email, etc) for a user managed by another terraform stack.
+---
+
+# sumologic_role
+
+Provides a way to retrieve Sumo Logic user details (id, email, etc) for a user
+managed by another terraform stack.
```
Comment (sumovishal) [inline website/docs/d/user.html.markdown:11]: "managed by another terraform stack" -> "managed outside of terraform"

File: sumologic/data_source_sumologic_user.go [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+			email := uemail.(string)
+			user, err = c.GetUserEmail(email)
+			if err != nil {
+				return fmt.Errorf("user with email address %s not found: %v", email, err)
+			}
+			if user == nil {
+				return fmt.Errorf("user with email address %s not found", email)
+			}
+		} else {
+			return errors.New("please specify either id or name")
```
Comment (sumovishal) [inline sumologic/data_source_sumologic_user.go:73]: "id or name" -> "id or email"

**Review by sumovishal — APPROVED** [review]
lgtm, some minor changes. Approving to unblock merging. Please address comments before merging.

File: website/docs/d/user.html.markdown (hunk 4 above)
Comment (emichaeli) [reply website/docs/d/user.html.markdown:11]: +1
Also, looks like the docs for `role` and `collector` data sources also say `"managed by another terraform stack"`. Maybe update those as well while you're at it? cc @troy-q-li

**Review by emichaeli — APPROVED** [review]
LGTM, nice 👍

**Comment by troy-q-li** [comment]
Address all the comments. Thanks for the review!

**Pull request merged**